### PR TITLE
RichCommands: Add Item & Edit Path

### LINKED ISSUE #11481: Generalizing Rich Command Usage
### Description

Thanks to @cinqmilleans' job, Files has now a cleaner way to handle commands across the solution. Rich commands now need to be implemented throughtout the whole solution.

This issue is used to keep track of which commands are made into Rich Commands.

### Concerned code

The following commands must be transformed into Rich Commands (See `BaseLayoutCommandsViewModel`, `ContextFlyoutItemHelper` and `ToolbarViewModel`):

- [x] Clear Selection (#11502)
- [x] Close Selected Tab (#11916)
- [x] Close Pane (#12002)
- [x] Compress Into Archive (#11572)
- [x] Compress Into Seven Zip (#11572)
- [x] Compress Into Zip (#11572)
- [x] Copy Item (#11504)
- [x] Copy Path Of Selected Items (#11774)
- [ ] Create Folder With Selection
- [x] Create New File
- [x] Create New Folder (#11493)
- [x] Create Shortcut (#11493)
- [x] Create Shortcut From Dialog (#11493)
- [x] Cut Item (#11504)
- [x] Decompress Archive (#11723)
- [x] Decompress Archive Here (#11723)
- [x] Decompress Archive To Child Folder (#11723)
- [x] Delete Item (#11504)
- [x] Duplicate Tab (#11596)
- [x] Empty Recycle Bin (#11415 )
- [x] Enter compact overlay (#11821)
- [x] Exit compact overlay (#11821)
- [x] Format Drive #11831
- [x] Gridview Size Decrease (#11678)
- [x] Gridview Size Increase (#11678)
- [x] Group Layer Sort (#11678)
- [x] Install Font (#11780)
- [x] Install Inf Driver (#11812)
- [x] Invert Selection (#11502)
- [x] Multi-Selection (#11502)
- [x] Navigate back (#11829)
- [x] Navigate forward (#11829)
- [x] Navigate up (#11829)
- [x] Navigate To Next Tab (#11916)
- [x] - [x] Navigate To Previous Tab (#11916)
- [ ] Open Directory In New Pane
- [ ] Open Directory In New Tab
- [x] Open File Location
- [ ] Open In New Window Item
- [x] Open In Terminal (#11445)
- [x] Open Item (#11667)
- [x] Open Item With Application Picker (#11667)
- [x] Open New Tab (#11596)
- [x] Open Parent Folder (#11667)
- [x] Open new pane (#12002)
- [x] Open settings #11846
- [x] Paste Items From Clipboard (#11670)
- [x] Pin Directory To Favorite
- [x] Play All (#12017)
- [x] Quick Look (#11771)
- [x] Redo (#12017)
- [x] Refresh #11807
- [x] Rename Item #11861
- [x] Reopen Tab (#11833)
- [x] Restore Item (#11472)
- [x] Restore Selection Recycle Bin (#11472)
- [x] Rotate Image Left (#11633)
- [x] Rotate Image Right (#11633)
- [x] Run As Admin (#11581)
- [x] Run As Another User (#11581)
- [x] Run With Powershell (#11914)
- [x] Search Unindexed Items (#12074)
- [x] Select All (#11502)
- [ ] Send to
- [x] Set As Lockscreen Background (#11563)
- [x] Set As Slideshow Item(#11563)
- [x] Set as Desktop Background(#11563)
- [x] Show Folder Properties (#11915)
- [x] Show Properties (#11915)
- [ ] Show/hide dot files
- [x] Share file (#11827)
- [x] Sidebar Pin Item (#11492)
- [x] Sidebar Unpin Item (#11492)
- [x] Toggle Preview Pane (#11578)
- [x] Toggle Sidebar #11847
- [x] Undo (#12017)
- [x] Unpin Directory From Favorite
- [x] Unpin Item From Start

### Gains

- A more cohesive code.
- The ability to call commands easily from anywhere in the code.

### Requirements

- Each of the commands previously cited have been transformed into Rich Commands.
- The previous implementations of the commands are cleaned up.

### Comments

_No response_

### PR DESCRIPTION
**Pull request by ferrariofilippo**
**Resolved / Related Issues**
- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Related #11481

**Notes**
- I also remove many `OnPropertyChanged` from `CurrentInstanceViewModel` since they were not influencing those properties
- I updated other `Create` actions to make sure they can be executed (they're not in recycle bin, ...)

**Validation**
How did you test these changes?
- [x] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [x] Did you remove any strings from the en-us resource file?
   - [x] Did you search the solution to see if the string is still being used? 
- [x] Did you implement any design changes to an existing feature?
   - [x] Was this change approved?
- [x] Are there any other steps that were used to validate these changes?
   1. Open app 
   2. Click new item button

### CODE REVIEW

**Comment by ferrariofilippo** [comment]
@d2dyno1 I noticed that we are preventing the user from creating files (but not folders) on MTP devices and FTP, is there any reason for this?

**Comment by cinqmilleans** [comment]
You can add KeyboardAcceleratorTextOverride on innerToolbar items (ToolbarNewFolderItem, ...) to display hotkeys. There are none but the user will be able to add his own.

File: src/Files.App/Strings/en-US/Resources.resw [hunk 1]
```diff
@@ -2875,4 +2875,7 @@
   <data name="CloseTab" xml:space="preserve">
     <value>Closes current tab</value>
   </data>
+  <data name="EditPath" xml:space="preserve">
+    <value>Edit éath</value>
```
Comment (d2dyno1) [inline src/Files.App/Strings/en-US/Resources.resw:2879]: Typo

**Comment by d2dyno1** [comment]
> @d2dyno1 I noticed that we are preventing the user from creating files (but not folders) on MTP devices and FTP, is there any reason for this?

I think a limitation prevented creating files in MTP devices but @gave92 would probably know the answer

**Comment by gave92** [comment]
> I noticed that we are preventing the user from creating files (but not folders) on MTP devices and FTP, is there any reason for this?

For MTP devices it was intentional -> see #7212. This is because calling CreateFileAsync() in a MTP folder does not work. For FTP perhaps it could be enabled.

File: specs/RichCommand/CommandList.md [hunk 2]
```diff
@@ -12,6 +12,7 @@ This is the list of all commands defined in `CommandCodes` enum except `None`.
 |                    | Search                              | Search                                    | Go to search box                                         | Ctrl+F, F3          |
 |                    | Redo                                | Redo                                      | Redo the last file operation                             | Ctrl+Y              |
 |                    | Undo                                | Undo                                      | Undo the last file operation                             | Ctrl+Z              |
+|                    | EditPath                            | Edit path                                 | Go to path bar                                           | Ctrl+L, Alt+D       |
```
Comment (yaira2) [inline specs/RichCommand/CommandList.md:15]:
```suggestion
|                    | EditPath                            | Edit path                                 | Focus path bar                                           | Ctrl+L, Alt+D       |

```

File: specs/RichCommand/CommandList.md [hunk 3]
```diff
@@ -24,6 +25,7 @@ This is the list of all commands defined in `CommandCodes` enum except `None`.
 |                    | DeleteItem                          | Delete                                    | Delete item(s)                                           | Delete, Ctrl+D      |
 |                    | DeletemeItemPermanently             | Delete permanently                        | Delete item(s) permanently                               | Shift+Delete        |
 |                    | CreateFolder                        | Folder                                    | Create new folder                                        |                     |
+|                    | AddItem                             | New                                       | Open the menu to create a new item                       | Ctrl+Shift+N        |
```
Comment (yaira2) [inline specs/RichCommand/CommandList.md:28]:
```suggestion
|                    | AddItem                             | New                                       | Create new item                       | Ctrl+Shift+N        |

```

File: src/Files.App/Strings/en-US/Resources.resw [hunk 4]
```diff
@@ -3178,6 +3178,12 @@
   <data name="ShowCheckboxesWhenSelectingItems" xml:space="preserve">
     <value>Show checkboxes when selecting items</value>
   </data>
+  <data name="EditPathDescription" xml:space="preserve">
+    <value>Go to path bar</value>
```
Comment (yaira2) [inline src/Files.App/Strings/en-US/Resources.resw:3182]: Can you update this to match the specs sheet?

Comment (ferrariofilippo) [reply src/Files.App/Strings/en-US/Resources.resw]: done

**Review by yaira2 — APPROVED** [review]
LGMT

**Pull request merged**